Realm/RLMUtil.mm

```diff
@@ -248,6 +249,50 @@ id RLMValidatedObjectForProperty(id obj, RLMProperty *prop, RLMSchema *schema) {
     return outArray;
 };
 
+NSArray *RLMCollectionValueForKey(NSString *key, RLMRealm *realm, RLMObjectSchema *objectSchema, size_t count, size_t (^indexGenerator)(size_t)) {
+    if (count == 0) {
+        return @[];
+    }
+    BOOL keyIsSelf = [key isEqualToString:@"self"];
+    NSMutableArray *results = [NSMutableArray arrayWithCapacity:count];
+    RLMObjectBase *accessor;
+    tightdb::Table *table = nullptr;
+    if (!keyIsSelf) {
```

Comment:
I don't see why we don't just separate out the case `keyIsSelf == YES` completely. Would make the code a bit easier to read without much code duplication.